deploy.sh

```diff
@@ -0,0 +1,131 @@
+#!/bin/bash
+set -o errexit
+set -o nounset
+set -o pipefail
+
+
+help() {
+    cat <<EOH
+    Usage: $0 [OPTIONS] <ENVIRONMENT>
+
+    Options:
+      -h        Show this help.
+      -c        Force refresh the certificates (only for new installations or
+                if they are expired).
+      -b        Also build the container image (locally only).
+      -v        Show verbose output.
+
+EOH
+}
+
+check_environment() {
+    # verify that the proper environment is passed
+
+    if [[ ! -d "deploy/$environment" || "$environment" =~ ^(base)$ ]]; then
+        echo "Unknown environment $environment, use one of:"
+        ls deploy/ | egrep -v '^(base)'
+        exit 1
+    fi
+}
+
+build_image() {
+    # build container image for dev or prod environments
+
+    local environment="${1?No environment passed}"
+    local do_build="${2?No build argument passed}"
+
+    if [[ "$do_build" == "yes" ]]; then
+        if [[ "$environment" == "devel" ]]
+        then
+            # creates the image on minikube's docker daemon
+            eval $(minikube docker-env)
+            docker build -t buildpack-admission:latest .
+        else
```

Comment:
I don't think we will want to run this script from `tools-docker-imagebuilder`, as it will try to deploy also the yamls and it will not be able to do so from there (also, there's only a tools docker imagebuilder, there's no toolsbeta one, nor toolsbeta specific docker registry).

Might be better to split the build and deploy scripts, or allow a build-only option.

Reply:
um I think this particular function exists after creating a build. the idea was that builds are only created in the docker builder. If user specifies -b, it only builds the image and pushes it to registry then exists with exit 0, if not it skips build and just deploys. But maybe thats too much logic

Reply:
@david-caro do you still advocate splitting build and deploy even though we already have the `-d` option? if so should `build.sh` follow the deploy standard too?

Reply:
Hey, I don't see a `-d` option, you mean the `-b`?

The issue is when trying to **only build** the container in the builder node, that does not have any access to the k8s cluster.

So we should find a way to just build the container without deploying it.

I'm ok if you prefer a different way that having a separated `build.sh` script, but we should be able to:
* deploy only by just running `deploy.sh` in production(tools)
* build only (in the builder node)
* do both locally when developing

Reply:
@david-caro I am still very confused about this. can we talk about this when you wake up? right now it does everything you listed above. we don't attempt to deploy when `deploy.sh -b` is ran. We only deploy when we only ran `deploy.sh <tools | toolsbeta |>`. We also do both building and deploying in dev environment

Reply:
@NdibeRaymond definitely! Let's have a chat :)